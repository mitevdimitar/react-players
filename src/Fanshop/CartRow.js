import React, {useState} from 'react';
import Grid from '@material-ui/core/Grid';
import Avatar from '@material-ui/core/Avatar';
import TextField from '@material-ui/core/TextField';
import AttachMoneyIcon from '@material-ui/icons/AttachMoney';
import HighlightOffIcon from '@material-ui/icons/HighlightOff';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
    row: {
      height: "50px",
      width: "90%",
      background: "lightgoldenrodyellow",
      marginBottom: "1px",
    },
    title: {
        paddingLeft: "30px",
        color: "#234465"
    },
    number: {
        width: "35px",
        color: "#234465"
    },
    deleteIcon: {
        color: "#234465",
        cursor: "pointer"
    },
  }));

function CartRow({product}) {
    const classes = useStyles();
    const [productQuantity, setProductQuantity] = useState(product.count)
    
    const quantityChange = (e) => {
        setProductQuantity(e.target.value)
    }
    return(
        <Grid container justify="center" alignItems="center" className={classes.row}>
            <Grid container justify="center" item xs={2}>
                <Avatar aria-label="logo" className={classes.avatar} src={require(`../img/${product.img}.png`)} />
            </Grid>
            <Grid item xs={6} className={classes.title}>{product.title}</Grid>
            <Grid container alignItems="center" item xs={1} className={classes.price}>{product.price} <AttachMoneyIcon style={{fontSize:"15px"}} /></Grid>
            <Grid container justify="center" item xs={2}>
            <TextField
                InputProps={{disableUnderline: true}}
                className={classes.number}
                id="standard-number"
                value={productQuantity}
                onChange={quantityChange}
                type="number"
            />
            </Grid>
            <Grid container alignItems="center" item xs={1} className={classes.deleteIcon}>
                <HighlightOffIcon />
            </Grid>
        </Grid>
    )
}

export default CartRow;