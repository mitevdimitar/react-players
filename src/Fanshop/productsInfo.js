/* 
https://www.iconfinder.com/icons/1243689/call_phone_icon
Creative Commons (Attribution 3.0 Unported);
https://www.iconfinder.com/Makoto_msk */

export const storeProducts = [
  {
    id: 1,
    title: "Premier League Ball",
    img: "../img/ball.png",
    price: 10,
    company: "PL",
    info:
      "Lorem ipsum dolor amet offal butcher quinoa sustainable gastropub, echo park actually green juice sriracha paleo. Brooklyn sriracha semiotics, DIY coloring book mixtape craft beer sartorial hella blue bottle. Tote bag wolf authentic try-hard put a bird on it mumblecore. Unicorn lumbersexual master cleanse blog hella VHS, vaporware sartorial church-key cardigan single-origin coffee lo-fi organic asymmetrical. Taxidermy semiotics celiac stumptown scenester normcore, ethical helvetica photo booth gentrify.",
    inCart: false,
    count: 0,
    total: 0
  },
  {
    id: 2,
    title: "Soccer Bag",
    img: "../img/bag.png",
    price: 16,
    company: "SAMSUNG",
    info:
      "Lorem ipsum dolor amet offal butcher quinoa sustainable gastropub, echo park actually green juice sriracha paleo. Brooklyn sriracha semiotics, DIY coloring book mixtape craft beer sartorial hella blue bottle. Tote bag wolf authentic try-hard put a bird on it mumblecore. Unicorn lumbersexual master cleanse blog hella VHS, vaporware sartorial church-key cardigan single-origin coffee lo-fi organic asymmetrical. Taxidermy semiotics celiac stumptown scenester normcore, ethical helvetica photo booth gentrify.",
    inCart: false,
    count: 0,
    total: 0
  },
  {
    id: 3,
    title: "Man Utd Jersey",
    img: "../img/jersey.png",
    price: 8,
    company: "htc",
    info:
      "Lorem ipsum dolor amet offal butcher quinoa sustainable gastropub, echo park actually green juice sriracha paleo. Brooklyn sriracha semiotics, DIY coloring book mixtape craft beer sartorial hella blue bottle. Tote bag wolf authentic try-hard put a bird on it mumblecore. Unicorn lumbersexual master cleanse blog hella VHS, vaporware sartorial church-key cardigan single-origin coffee lo-fi organic asymmetrical. Taxidermy semiotics celiac stumptown scenester normcore, ethical helvetica photo booth gentrify.",
    inCart: false,
    count: 0,
    total: 0
  }
];

export const detailProduct = {
  id: 1,
  title: "Google Pixel - Black",
  img: "img/product-1.png",
  price: 10,
  company: "google",
  info:
    "Lorem ipsum dolor amet offal butcher quinoa sustainable gastropub, echo park actually green juice sriracha paleo. Brooklyn sriracha semiotics, DIY coloring book mixtape craft beer sartorial hella blue bottle. Tote bag wolf authentic try-hard put a bird on it mumblecore. Unicorn lumbersexual master cleanse blog hella VHS, vaporware sartorial church-key cardigan single-origin coffee lo-fi organic asymmetrical. Taxidermy semiotics celiac stumptown scenester normcore, ethical helvetica photo booth gentrify.",
  inCart: false,
  count: 0,
  total: 0
};
