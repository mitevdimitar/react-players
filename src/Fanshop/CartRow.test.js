import React from 'react';
import renderer from 'react-test-renderer';
import CartRow from './CartRow';
import { storeProducts } from "./productsInfo";

test('product', () => {
    const product = storeProducts[0];
    expect(product).toBeDefined();
});

test('Cart row should render corectly', () => {
  const component = renderer.create(
    <CartRow product={storeProducts[0]}></CartRow>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});