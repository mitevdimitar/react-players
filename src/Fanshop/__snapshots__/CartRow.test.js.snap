// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart row should render corectly 1`] = `null`;
